src/test/unit/algorithms/SpatialPoolerTest.cpp

```diff
@@ -1064,7 +1067,9 @@ TEST(SpatialPoolerTest, testCalculateOverlap) {
 
   for (UInt i = 0; i < numTrials; i++) {
     vector<UInt> overlaps;
-    sp.calculateOverlap_(inputs[i], overlaps);
+    SDR input({numInputs});
+    input.setDense(SDR_dense_t(inputs[i], inputs[i] + numInputs));
```

Comment:
use SDR.at(), resp should we have operator[] to do that?

Reply:
I'm confused about what is this doing?

Reply:
`input.setDense(SDR_dense_t(inputs[i], inputs[i] + numInputs));`

This makes a new SDR_dense_t aka vector<Byte>.  It is initialized with the array `inputs[i]`, and immediately copied into the SDR by input.setDense.  `inputs` contains 5 separate tests worth of data, and it's iterated through by index `i`.  Does this help?

Reply:
yes, sorry, I thought `intputs[i]` is an index, not that input is an array of arrays. Makes sense.